### PR TITLE
extend timeout and more logging

### PR DESCRIPTION
**Pull request by analogrelay**
Also, we were not disposing the (disposable) WebSocket in the Autobahn Test app.

NOTE: use https://github.com/aspnet/WebSockets/pull/183/files?w=1 to review, indentation changes :)

/cc @ryanbrandenburg

### CODE REVIEW

File: test/AutobahnTestApp/Startup.cs [hunk 1]
```diff
@@ -22,8 +22,10 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
                 if (context.WebSockets.IsWebSocketRequest)
                 {
                     logger.LogInformation("Received WebSocket request");
-                    var webSocket = await context.WebSockets.AcceptWebSocketAsync();
-                    await Echo(webSocket);
+                    using (var webSocket = await context.WebSockets.AcceptWebSocketAsync())
```
Comment (ryanbrandenburg) [inline test/AutobahnTestApp/Startup.cs]: `using` always makes me happy.

File: test/Microsoft.AspNetCore.WebSockets.ConformanceTest/AutobahnTests.cs [hunk 2]
```diff
@@ -7,13 +7,16 @@
...
 using Microsoft.AspNetCore.WebSockets.ConformanceTest.Autobahn;
+using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Logging.Testing;
 using Xunit.Abstractions;
 
 namespace Microsoft.AspNetCore.WebSockets.ConformanceTest
 {
     public class AutobahnTests : LoggedTest
     {
+        private static readonly TimeSpan TestTimeout = TimeSpan.FromMinutes(10);
```
Comment (ryanbrandenburg) [inline test/Microsoft.AspNetCore.WebSockets.ConformanceTest/AutobahnTests.cs]: I'm not a fan of bumping up timeout when we run into spurrious test, I think it hides real bugs. I think the question to ask ourselves is "would a user think this timeout is a reasonable timeframe for this scenario?". If the answer is no then the timeout is probably too high, and I'm going to guess that 10 minutes (and maybe 5) is too high.

File: test/Microsoft.AspNetCore.WebSockets.ConformanceTest/AutobahnTests.cs [hunk 3]
```diff
@@ -44,28 +48,31 @@ public AutobahnTests(ITestOutputHelper output) : base(output)
                     .ExcludeCase("9.*", "12.*", "13.*");
 
                 var cts = new CancellationTokenSource();
-                cts.CancelAfter(TimeSpan.FromMinutes(5)); // These tests generally complete in just over 1 minute.
+                cts.CancelAfter(TestTimeout); // These tests generally complete in just over 1 minute.
 
-                AutobahnResult result;
-                using (var tester = new AutobahnTester(loggerFactory, spec))
+                using (cts.Token.Register(() => logger.LogError("Test run is taking longer than expected duration of {timeoutMinutes:0.00} minutes. Aborting...", TestTimeout.TotalMinutes)))
```
Comment (ryanbrandenburg) [inline test/Microsoft.AspNetCore.WebSockets.ConformanceTest/AutobahnTests.cs]: You say above that the expected duration is 1 minute, so `Test run is taking longer than maximum duration of...`.

File: test/Microsoft.AspNetCore.WebSockets.ConformanceTest/AutobahnTests.cs [hunk 4]
```diff
@@ -44,28 +48,31 @@ public AutobahnTests(ITestOutputHelper output) : base(output)
...
                 {
-                    await tester.DeployTestAndAddToSpec(ServerType.Kestrel, ssl: false, environment: "ManagedSockets", cancellationToken: cts.Token);
-
-                    // Windows-only WebListener tests, and Kestrel SSL tests (due to: https://github.com/aspnet/WebSockets/issues/102)
-                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    AutobahnResult result;
+                    using (var tester = new AutobahnTester(loggerFactory, spec))
                     {
-                        await tester.DeployTestAndAddToSpec(ServerType.Kestrel, ssl: true, environment: "ManagedSockets", cancellationToken: cts.Token);
+                        await tester.DeployTestAndAddToSpec(ServerType.Kestrel, ssl: false, environment: "ManagedSockets", cancellationToken: cts.Token);
```
Comment (ryanbrandenburg) [inline test/Microsoft.AspNetCore.WebSockets.ConformanceTest/AutobahnTests.cs]: Why drop ssl?

**Review by ryanbrandenburg — APPROVED** [review]
:shipit: after the minor text change and either lowering the timeout back down or explaining why it would be ok to be that high.

File: test/Microsoft.AspNetCore.WebSockets.ConformanceTest/AutobahnTests.cs (hunk 4 above)
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.WebSockets.ConformanceTest/AutobahnTests.cs]: He didn't, github formatting issue

Comment (ryanbrandenburg) [inline test/Microsoft.AspNetCore.WebSockets.ConformanceTest/AutobahnTests.cs]: Oh I see. Yep, ignore this one.

**Comment by anurse** [comment]
:up: :date: Reacted to PR feedback, also added a "pinger" that sends standard HTTP requests to the server while the test is running so we can identify if the server or client is hanging, when they hang.

File: test/AutobahnTestApp/Startup.cs (hunk 1 above)
Comment (Tratcher) [inline test/AutobahnTestApp/Startup.cs]: Can you move this dispose into the middleware?
https://github.com/aspnet/WebSockets/issues/182#issuecomment-306214086

**Pull request merged**